<launch>
  <arg name="filter" default="phd" />
  <arg name="sim" default="false" />
  <arg name="num_drones" default="2" />
  <arg name="config_file" default="params_snapdragon_flight.yaml" />
  
  <arg name="start_nodelet_manager" default="false" />
  <arg name="nodelet_manager_name" default="nodelet_manager_multi_tracking" />
  <param name="filter" value="$(arg filter)" />
  <param name="/use_sim_time" value="false" />

  <!-- Related Nodelets-->
  <group ns="/hummingbird0">
    <node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name)" args="manager" output="screen" if="$(arg start_nodelet_manager)" />
    <node pkg="multi_robot_tracking" type="multi_robot_tracking_node" name="multi_tracking" required="true" output="screen">
      <rosparam file="$(find multi_robot_tracking)/config/$(arg config_file)" unless="$(arg sim)"/>
      <rosparam file="$(find multi_robot_tracking)/config/params_flightmare_sim.yaml" if="$(arg sim)"/>
      <param name="filter" value="$(arg filter)" />
      <param name="num_drones" value="$(arg num_drones)" />
    </node>
  </group>

  <!-- <node pkg="rosbag" type="play" name="player" output="screen" args="/home/vivek/ws/NYU/ARPL/RAL2021/data/rosbag/drone_2d_3drones_imu.bag" required="true" if="$(arg sim)"/> -->

</launch>